kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -992,6 +1004,75 @@ saveSession path =
     seqUnlines :: Seq String -> String
     seqUnlines = unlines . toList
 
+savePartialProof
+    :: forall m claim
+    .  MonadState (ReplState claim) m
+    => MonadWriter ReplOutput m
+    => MonadIO m
+    => Claim claim
+    => From claim (TermLike Variable)
+    => Maybe Natural
+    -> FilePath
+    -> m ()
+savePartialProof maybeNatural file = do
+    currentClaim <- Lens.use (field @"claim")
+    currentIndex <- Lens.use (field @"claimIndex")
+    claims <- Lens.use (field @"claims")
+    maybeConfig <- getConfigAt maybeNode
+    case maybeConfig of
+        Nothing -> putStrLn' "Invalid node!"
+        Just (_, currentProofState) -> do
+            let config = unwrapConfig currentProofState
+                newClaim = createClaim currentClaim config
+                newTrustedClaims =
+                    makeTrusted <$> removeAt currentIndex claims
+                newDefinition =
+                    createNewDefinition
```

Comment:
I think the original spec module may contain things like symbol definitions, it would be nice to preserve them.

Reply:
I thought only claim definitions could appear in the spec module, I asked @traiansf about this, but maybe I misunderstood.

Reply:
Ok, he should know more about this than me.